### ISSUE TITLE
:lint-as section for component functions called with hiccup

### ISSUE THREAD

**Issue by sirmspencer** [clj-kondo/clj-kondo#1751]
To upvote this issue, give it a thumbs up. See [this list](https://github.com/clj-kondo/clj-kondo/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) for the most upvoted issues.

See this old ticket for reference https://github.com/clj-kondo/clj-kondo/issues/25

**Is your feature request related to a problem? Please describe.**

When calling a component function with hiccup, it is in a vector instead of a function, so it doesn't lint. 

```
(defn my-component []
  [:div "Hello"])

[:div [my-component 1 2 3]] ;; <- should be arity error
```

**Describe the solution you'd like**
In the old ticket there was concern for false positives.  What about adding section in or similar to :lint-as where we can define which component functions are worth linting as functions.  In my project, I would mostly include larger reused components.

**Comment by borkdude** [comment]
An alternative would be to write a macro:

```
(defmacro $ [fn & args]
  (into [fn] args))
```

and then use this macro in your code for functions you care about for arity errors:

```
($ my-component 1 2 3)
```

which is translated to `[my-component 1 2 3]` for reagent. At the same time you configure clj-kondo with a hook for this macro as:

```
(defmacro $ [fn & args]
  (list* fn args))
```

so that clj-kondo will actually see this as a function call.

**Comment by sirmspencer** [comment]
> An alternative would be to write a macro:
> 
> ```
> (defmacro $ [fn & args]
>   (into [fn] args))
> ```
> 
> and then use this macro in your code for functions you care about for arity errors:
> 
> ```
> ($ my-component 1 2 3)
> ```
> 
> which is translated to `[my-component 1 2 3]` for reagent. At the same time you configure clj-kondo with a hook for this macro as:
> 
> ```
> (defmacro $ [fn & args]
>   (list* fn args))
> ```
> 
> so that clj-kondo will actually see this as a function call.

Thats nifty for sure.  The only downside is the need to import the macro everywhere.
